### PR TITLE
Fix memory leak in TorrentChecker

### PR DESCRIPTION
**Pull request by ichorid**
`session_stop_task_list` was used to track the coroutines for stopping tracker sessions. The list was never emptied which resulted in a memory leak of about 50 MB/hour.

This PR deprecates `session_stop_task_list` in favour of TaskManager, which stops the leak and makes the code cleaner.

### CODE REVIEW

File: src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py [hunk 1]
```diff
@@ -85,15 +82,6 @@ def __init__(self, session):
 
         await self.shutdown_task_manager()
 
-        # kill all the tracker sessions.
-        # Wait for the defers to all have triggered by using a DeferredList
-        for tracker_url in self._session_list.keys():
-            for session in self._session_list[tracker_url]:
-                self.session_stop_task_list.append(session.cleanup())
```
Comment (egbertbouman) [inline src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py]: Doesn't removing this mean that we don't shutdown the TaskManager of each individual session?

Comment (ichorid) [reply src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py]: Each HTTPSession cleans itself on cancel and when it is done:
https://github.com/Tribler/tribler/blob/8fcba201280effb89dff424b557bbfa240abf16f/src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py#L316

Comment (egbertbouman) [reply src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py]: I could be wrong, but looking at `connect_to_tracker` it seems that if there is a `CancelledError`, it doesn't call `clean_session`.
https://github.com/Tribler/tribler/blob/8fcba201280effb89dff424b557bbfa240abf16f/src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py#L136-L147

Comment (ichorid) [reply src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py]: You're right, I'll fix it.

Comment (ichorid) [reply src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py]: added `.clean_session` call to `CancelledError`

**Pull request merged**